tmail-backend/integration-tests/rate-limiter/rate-limiter-integration-tests-common/src/main/scala/com/linagora/tmail/integration/RateLimitingPlanIntegrationContract.scala

```diff
@@ -0,0 +1,179 @@
+import org.apache.james.mailets.configuration.Constants.{LOCALHOST_IP, PASSWORD, awaitAtMostOneMinute}
+import org.apache.james.mailrepository.api.MailRepositoryUrl
+import org.apache.james.modules.protocols.{ImapGuiceProbe, SmtpGuiceProbe}
+import org.apache.james.utils.TestIMAPClient.INBOX
+import org.apache.james.utils.{DataProbeImpl, MailRepositoryProbeImpl, SMTPMessageSender, TestIMAPClient, WebAdminGuiceProbe}
+import org.apache.james.webadmin.WebAdminUtils
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+import java.util.UUID
+
+object RateLimitingPlanIntegrationContract {
+  val DOMAIN: Domain = Domain.of("domain.tld")
+  val SENDER1: Username = Username.fromLocalPartWithDomain("sender1", DOMAIN)
+  val SENDER2: Username = Username.fromLocalPartWithDomain("sender2", DOMAIN)
+  val RECIPIENT1: Username = Username.fromLocalPartWithDomain("recipient1", DOMAIN)
+  val RECIPIENT2: Username = Username.fromLocalPartWithDomain("recipient2", DOMAIN)
+  val ERROR_REPOSITORY: MailRepositoryUrl = MailRepositoryUrl.from("cassandra://var/mail/error/")
+}
+
+trait RateLimitingPlanIntegrationContract {
+
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString())
+      .addUser(SENDER1.asString(), PASSWORD)
+      .addUser(SENDER2.asString(), PASSWORD)
+      .addUser(RECIPIENT1.asString(), PASSWORD)
+      .addUser(RECIPIENT2.asString(), PASSWORD)
+
+    requestSpecification = WebAdminUtils.buildRequestSpecification(server.getProbe(classOf[WebAdminGuiceProbe]).getWebAdminPort)
+      .build
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails()
+  }
+
+  private def messageSender(server: GuiceJamesServer): SMTPMessageSender =
+    new SMTPMessageSender(DOMAIN.asString())
+      .connect(LOCALHOST_IP, server.getProbe(classOf[SmtpGuiceProbe]).getSmtpPort)
+
+  private def testImapClient(server: GuiceJamesServer): TestIMAPClient =
+    new TestIMAPClient().connect(LOCALHOST_IP, server.getProbe(classOf[ImapGuiceProbe]).getImapPort)
+
+  private def createNewPlan(): String = {
+    val json: String =
+      """{
+        |  "transitLimits": [
+        |    {
+        |      "name": "receivedMailsPerHour",
+        |      "periodInSeconds": 3600,
+        |      "count": 1,
+        |      "size": 2048
+        |    }
+        |  ],
+        |  "relayLimits": [
+        |    {
+        |      "name": "relayMailsPerHour",
+        |      "periodInSeconds": 3600,
+        |      "count": 1,
+        |      "size": 2048
+        |    }
+        |  ],
+        |  "deliveryLimits": [
+        |    {
+        |      "name": "deliveryMailsPerHour",
+        |      "periodInSeconds": 3600,
+        |      "count": 1,
+        |      "size": 2048
+        |    }
+        |  ]
+        |}""".stripMargin
+
+    given().basePath("/rate-limit-plans/" + UUID.randomUUID().toString)
```

Comment:
IMO `UUID.randomUUID().toString` => `"somePlanName"`

Just a test, but this could lead to some misunderstanding about the route parameter.